echidna/core/scale.py

```diff
@@ -0,0 +1,157 @@
+import echidna.core.spectra as spectra
+import numpy
+
+
+class Scale(object):
+    """ A class for scaling the parameter space of a spectra.
+
+    Attributes:
+      _scale_factor (float): The factor you want to scale a parameter by.
+    """
+
+    def __init__(self):
+        """ Initialise the Scale class.
+        """
+        self._scale_factor = 1.
+
+    def get_scale_factor(self):
+        """ Returns the scale factor.
+
+        Returns:
+          float: The scale factor.
+        """
+        return self._scale_factor
+
+    def set_scale_factor(self, scale_factor):
+        """ Sets the scale factor.
+
+        Args:
+          scale_factor (float): Value you wish to set the scale factor to.
+
+       Raises:
+         ValueError: If scale_factor is zero or below.
+        """
+        if scale_factor <= 0.:
+            raise ValueError("Scale factor must be positive and non-zero.")
+        self._scale_factor = scale_factor
+
+    def scale(self, spectrum, dimension, **kwargs):
+        """ Scales a given spectrum's dimension.
+
+        Args:
+          spectrum (float): The spectrum you want to scale.
+          dimension (string): The dimension of the spectrum you want to scale.
+          kwargs (dict): To passed to the interpolation function in
+            :class:`echidna.core.spectra.Spectra`
+
+        Returns:
+          :class:`echidna.core.spectra.Spectra`: The scaled spectrum.
+        """
+        prescale_sum = spectrum.sum()
+        interpolation = spectrum.interpolate1d(dimension, **kwargs)
+        sf = self.get_scale_factor()
+        scaled_spec = spectra.Spectra(spectrum._name+"_sf" +
+                                      str(sf),
+                                      spectrum._num_decays,
+                                      spectrum.get_config())
+        n_dim = len(spectrum._data.shape)
+        axis = spectrum.get_config().get_index(dimension)
+        par = spectrum.get_config().get_par(dimension)
+        low = par._low
+        n_bins = par._bins
+        step = par.get_width()
+        for bin in range(n_bins):
+            x = par.get_bin_centre(bin)
+            if x/sf < low or x/sf > par._high:
+                continue  # Trying to scale values outside range (Unknown)
+            y = interpolation(x/sf)
```

Comment:
So I think @arushanova was referring to the the interpolation part. You could pass a `use_interpolation=True` to the method to invoke this behaviour, otherwise just assume a uniform distribution of events within the bin, i.e. y = current bin content at x/sf. Having thought about this some more though, the reason for adding this option in would be to speed up the scale process, should the user wish, but I think we actually need to see what the trade off is between speed an accuracy. 

This is possibly a separate issue.